services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/query/Queries.java

```diff
@@ -0,0 +1,151 @@
+package org.sagebionetworks.repo.manager.table.query;
+
+import java.util.Optional;
+
+import org.sagebionetworks.repo.model.table.Query;
+import org.sagebionetworks.repo.model.table.QueryOptions;
+import org.sagebionetworks.table.cluster.SchemaProvider;
+import org.sagebionetworks.table.cluster.TranslationDependencies;
+import org.sagebionetworks.table.cluster.description.IndexDescription;
+
+public class Queries {
+
+	private final MainQuery mainQuery;
+	private final FacetQueries facetQueries;
+	private final CountQuery countQuery;
+	private final SumFileSizesQuery sumFileSizesQuery;
+
+	private Queries(String startingSql, SchemaProvider schemaProvider, IndexDescription indexDescription, Long userId,
+			Long maxBytesPerPage, Query query, QueryOptions options, Long maxRowsPerCall) {
+		CachingSchemaProvider providerCache = new CachingSchemaProvider(schemaProvider);
+		TranslationDependencies dependencies = TranslationDependencies.builder().setSchemaProvider(providerCache)
+				.setIndexDescription(indexDescription).setUserId(userId).build();
+
+		mainQuery = MainQuery.builder().setStartingSql(startingSql).setDependencies(dependencies)
```

Comment:
I would suggest moving the business logic used to build this object in the build method itself and keep this (private) constructor anemic passing the builder instance itself.